api/v1/verticadb_webhook.go

```diff
@@ -1011,6 +1013,20 @@ func (v *VerticaDB) checkImmutableDeploymentMethod(oldObj *VerticaDB, allErrs fi
 	return allErrs
 }
 
+// checkImmutableNMADeploymentMethod will check if the nma deployment type is changing from
+// sidecar to monolithic
+func (v *VerticaDB) checkImmutableNMADeploymentMethod(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
```

Comment:
I am not sure if we want any kind of check but I added this because I thought it should not be possible to go from sidecar to monolithic